terraform-aws-github-runner/modules/runners/lambdas/runners/src/scale-runners/cache.ts

```diff
@@ -278,36 +278,77 @@ export async function redisLocked<T>(
   throw new Error(`Could not acquire lock for ${nameSpace}-${key}`);
 }
 
-export async function getExperimentValue(experimentKey: string, defaultValue: number): Promise<number> {
-  await startupRedisPool();
-  if (!redisPool) throw Error('Redis should be initialized!');
+export async function getExperimentValueStr(experimentKey: string, defaultValue: string): Promise<string> {
+  return locallyCached('EXPERIMENT', experimentKey, 60, async (): Promise<string> => {
+    await startupRedisPool();
+    if (!redisPool) throw Error('Redis should be initialized!');
 
-  const queryKey = `${Config.Instance.environment}.EXPERIMENT.${experimentKey}`;
+    const queryKey = `${Config.Instance.environment}.EXPERIMENT.${experimentKey}`;
 
-  console.debug(`Checking for experiment ${experimentKey} (${queryKey})`);
-  try {
-    const experimentValue: string | undefined | null = await redisPool.use(async (client: RedisClientType) => {
-      return await client.get(queryKey);
-    });
-
-    if (experimentValue !== undefined && experimentValue !== null) {
-      const numValue = Number(experimentValue);
-      if (!isNaN(numValue)) {
-        console.debug(`Found experiment ${queryKey} with value ${numValue}`);
-        return numValue;
+    console.debug(`Checking redis entry for experiment ${experimentKey} (${queryKey})`);
+    try {
+      const experimentValue: string | undefined | null = await redisPool.use(async (client: RedisClientType) => {
```

Comment:
Prob should have some type validation of the result type here as well.  IIRC you can store much more than just strings as a redis key's value (e.g. lists, sets, and more)

Maybe have a separate `getExperimentValueRaw()` method and then have `getExperimentValueStr()` and `getExperimentValueInt()` to validate the returned type was as expected?